src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java

```diff
@@ -17,138 +18,150 @@
+    @FindBy(css = ".local-nav-menu > li")
+    private List<WebElement> navList;
+    @FindBy(css = ".back")
+    private WebElement backChevron;
+    @FindBy(css = ".random-article-link")
+    private WebElement randomPageButton;
+    @FindBy(css = ".overlay")
+    private WebElement overlay;
+    @FindBy(css = ".local-wikia-search")
+    private WebElement resultField;
+    @FindBy(css = ".local-nav-menu")
+    private WebElement menuField;
+    @FindBy(css = ".nav")
+    private WebElement searchButton;
+    @FindBy(css = "nav.side-nav")
+    private WebElement menuView;
+    @FindBy(xpath = "//span[contains(.,'Sorry, we could')]")
+    private WebElement sorrySpan;
+    @FindBy(css = "div.side-nav-toggle")
+    private WebElement openListIcon;
+
+
+    public NavigationSideComponentObject(WebDriver driver) {
+        super(driver);
+    }
+
+    public void clickSearchField() {
+        wait.forElementVisible(searchInput);
+        searchInput.click();
+    }
+
+    public void clickSearchButton() {
+        wait.forElementVisible(searchButton);
+        searchButton.click();
+    }
+
+    public void clickBackChevron() {
+        wait.forElementVisible(backChevron);
+        backChevron.click();
+    }
+
+    public void clickCancelButton() {
+        wait.forElementVisible(cancelSearchCaption);
+        cancelSearchCaption.click();
+    }
+
+    public void clickSuggestion(int index) {
+        wait.forElementVisible(searchSuggestion);
+        searchSuggestion.click();
+    }
+
+    public void clickNavListElement(int index) {
+        wait.forElementVisible(navList.get(index));
+        navList.get(index).click();
+    }
+
+    public void clickOverlay() {
+        wait.forElementVisible(overlay);
+        tapOnElement(overlay);
+    }
+
+    public ArticlePageObject clickRandomPageButton() {
+        wait.forElementVisible(randomPageButton);
+        randomPageButton.click();
+
+        return new ArticlePageObject(driver);
+    }
+
+    public boolean isSuggestionListDisplayed() {
+        try {
+            wait.forElementVisible(searchSuggestion, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isSorryInfoDisplayed() {
+        try {
+            wait.forElementVisible(sorrySpan, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isNavMenuVisible() throws WebDriverException {
+        if (menuView.getAttribute("class") == null) {
+            throw new WebDriverException("Expected String but got null");
+        }
+        return !menuView.getAttribute("class").contains("collapsed");
+    }
+
+    public boolean isBackLinkDisplayed() {
+        try {
+            wait.forElementVisible(backChevron, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)